### DISCUSSION TITLE
SpatialDWLS analysis

### DISCUSSION THREAD

**Discussion by vwtlin** [giotto-suite/Giotto#576] [Q&A] [answered]
Hi Giotto team,

Firstly thanks very much for such a fantastic tool for spatial analysis. I am analysing Visium FFPE data and examining the co-existence of cell types within the same spots. The spatialDWLS analysis is awesome to visualise the cell-type percentages. I am wondering if it is possible to extract the matrix/data that generate this deconvolution plot as I guess that could be used to make some quantitative analysis.

Cheers,
Weitao

**Comment by RubD** [comment, marked as answer]
Hi @vwtlin 

Assuming you're using our Giotto `suite` or `suite_dev` (development) branch you should be able to use the `getSpatialEnrichment` function to extract the matrix with the DWLS results.

Let us know if it works, thanks!

Ruben

**Reply by vwtlin** [reply]
Hi @RubD,

Thanks very much for your reply. It is delayed to try this function as I was using Giotto not Giotto Suite. I didn't realize the difference before. The `getSpatialEnrichment` function works! Also notice that the [documentation ](https://giottosuite.readthedocs.io/en/latest/subsections/md_rst/get_spatial_enrichment.html)on your website hasn't been updated.

Btw, do you mind if I ask another question in DWLS analysis? I reckon that the number of signature genes in each cell type affects deconvolution. What would be an optional number of genes to be provided?

Cheers,
Weitao

**Comment by gcyuan** [comment]
Hi @vwtlin 

Good question. We have not systematically evaluated the impact of the number of signature genes on deconvolution results. The optimal number is likely to be dataset dependent. If you are interested to test on your own, you could use the default setting as the starting point and vary the number of genes to see how performance changes. Let us know if you have any further questions.

Best,
GC

**Reply by vwtlin** [reply]
Hi @gcyuan 

Thanks very much for your advice. I will go and try. 

I do have some questions.

1. I ran the DWLS analysis this weekend and the result seems ok to me now. I am wondering why there are some empty spots on the left. Those spots are covered by tissue. I suspect the DWLS analysis could not predict the composition of those spots. Could you advise if I am correct or not?
2. In the original tutorial, the Giotto object could be sub-set based on the coordinate of the image. Is there a way to interactively visualize the spot barcodes so that I could isolate the spots of interest for further downstream?

Cheers,
Weitao

![27-spatDeconvPlot](https://user-images.githubusercontent.com/44833405/224610450-353578c6-02d0-4fcf-9a3d-fea44b442824.png)

**Reply by gcyuan** [reply]
Hi @vwtlin  I am not exactly sure what the problem is. The figure you attached looks fine to me. Could you provide more detail? If your main concern is that some of the cells covered by tissue are removed, then this is because they are removed during the filtering step because the data quality is not sufficiently good. You should be able to find the outcome associated with each spot barcode using the method Ruben described earlier.

**Reply by vwtlin** [reply]
Hi @gcyuan, thanks for your reply. I don't think the spots were removed after QC step as they could still be seen in the leiden clustering and gene expression plot. That is why I thought they failed in the DWLS step. 
![10-spatPlot2D](https://user-images.githubusercontent.com/44833405/225491442-af4a606a-a46f-4776-ad8c-0b12902548ea.png)
![16-spatFeatPlot2D](https://user-images.githubusercontent.com/44833405/225491938-a033e072-e6f7-4fa3-8ad5-6ed794dcf8cf.png)

Would you mind if you could comment on whether there is some way to visualize the spot barcodes?

Cheers,
Weitao

**Comment by gcyuan** [comment]
This is odd. Have you checked the spatialDWLS output manually? What are the estimated cell-type fractions corresponding to  these missing spots?

**Reply by vwtlin** [reply]
Hi @gcyuan, I was thinking to check but not sure which spot barcodes to look for. That is why I was asking about the way to check/visualise the barcodes.

**Comment by RubD** [comment]
Hi @vwtlin 
If you've switched to GIotto Suite and are using `getSpatialEnrichment` you can inspect that data.table output for NA values. I believe that's what probably causing these spots to go missing. There is an internal function to convert the data.table to a standard R matrix `Giotto:::dt_to_matrix()` which could you to inspect the results as a matrix.

**Reply by vwtlin** [reply]
Hi @RubD, 

I had a look at the DWLS output matrix and I didn't see any NA value. Instead, I found that all the proportion values were 0 in all cell types for some spots. Therefore, I suspect the deconvolution failed in those spots.

Also, thanks for showing the function to convert data.table to matrix. I was thinking to ask how to export the count matrix for infercnv analaysis. 

Cheers,
Weitao

**Reply by gcyuan** [reply]
Hi @vwtlin , this is odd. It shouldn’t happen. Without seeing the data, it is difficult for us to speculate what went wrong. If you are willing to share the data with us we will be happy to help you troubleshoot. Feel free to withhold any confidential information.
